core/src/main/java/cucumber/runtime/arquillian/feature/FeaturesManager.java

```diff
@@ -0,0 +1,370 @@
+package cucumber.runtime.arquillian.feature;
+
+import cucumber.api.CucumberOptions;
+import cucumber.runtime.arquillian.api.Features;
+import cucumber.runtime.arquillian.lifecycle.CucumberLifecycle;
+import cucumber.runtime.io.FileResource;
+import cucumber.runtime.io.MultiLoader;
+import cucumber.runtime.io.Resource;
+import cucumber.runtime.io.ResourceLoader;
+import cucumber.runtime.io.ZipResource;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import static cucumber.runtime.arquillian.shared.IOs.dump;
+import static cucumber.runtime.arquillian.shared.IOs.slurp;
+import static java.util.Arrays.asList;
+
+public final class FeaturesManager {
```

Comment:
Seems about right to me, I called it mapper because the single public operation of the class is the construction of the Map, but finder is more closely to what it actually does